html/semantics/embedded-content/media-elements/src_object_blob.html

```diff
@@ -6,26 +6,20 @@
 <script src="/resources/testharnessreport.js"></script>
 <video></video>
 <script>
-  async_test(function(t) {
-    t.step(function() {
-      fetch(getVideoURI('/media/movie_5'))
-        .then(function(response) {
-          return response.blob();
-        })
-        .then(function(blob) {
-          let video = document.querySelector("video");
-          video.srcObject = blob;
-          video.addEventListener('ended', function() {
-            t.done();
-          });
-          video.play().catch(function(error) {
-            assert(false, error);
-          });
-        })
-        .catch(function(error) {
-          assert(false, error);
-        });
+  let video = document.querySelector("video");
+  promise_test(async t => {
+    const blob = await fetch(getVideoURI('/media/movie_5'))
+          .then(r => r.blob());
+    try {
+      video.srcObject = blob;
+    } catch (error) {
+      assert_unreached(error);
+      return t.done();
+    }
+    video.addEventListener('ended', function() {
+      t.done();
     });
+    video.play();
```

Comment:
Since you don't seem to return anything, this test will always pass (unless it fails during the await bit, but that seems unlikely). You'll need to return a promise that you resolve from the ended listener. `t.done()` is not appropriate for promise tests as far as I know.

Reply:
doh, indeed -  in fact, the failure that is interesting (and the one I was focusing in in my rewrite) is in try/catch - I've fixed the promise flow